packages/common/src/ee/Ai/schemas.ts

```diff
@@ -183,39 +183,59 @@ const dateFilterSchema = z.union([
     }),
 ]);
 
-const filterRuleSchema = z
-    .object({
-        type: z.enum(['or', 'and']).describe('Type of filter group operation'),
-        target: z.object({
-            fieldId: fieldIdSchema,
-            type: FieldTypeSchema,
-        }),
-        rule: z.union([
-            booleanFilterSchema.describe('Boolean filter'),
-            stringFilterSchema.describe('String filter'),
-            numberFilterSchema.describe('Number filter'),
-            dateFilterSchema.describe('Date filter'),
-        ]),
-    })
-    .transform(
-        (data): FilterRule => ({
-            id: uuid(),
-            target: data.target,
-            operator: data.rule.operator,
-            values: 'values' in data.rule ? data.rule.values : [],
-            ...('settings' in data.rule
-                ? { settings: data.rule.settings }
-                : {}),
-        }),
-    );
+/**
+ * Raw schema for filter rules that are passed to the AI.
+ */
+const filterRuleSchema = z.object({
+    type: z.enum(['or', 'and']).describe('Type of filter group operation'),
+    target: z.object({
+        fieldId: fieldIdSchema,
+        type: FieldTypeSchema,
+    }),
+    rule: z.union([
+        booleanFilterSchema.describe('Boolean filter'),
+        stringFilterSchema.describe('String filter'),
+        numberFilterSchema.describe('Number filter'),
+        dateFilterSchema.describe('Date filter'),
+    ]),
+});
+
+/**
+ * Transformed schema for filter rules that are passed to the query.
+ */
+const filterRuleSchemaTransformed = filterRuleSchema.transform(
+    (data): FilterRule => ({
+        id: uuid(),
+        target: data.target,
+        operator: data.rule.operator,
+        values: 'values' in data.rule ? data.rule.values : [],
+        ...('settings' in data.rule ? { settings: data.rule.settings } : {}),
+    }),
+);
+
+/**
+ * Raw schema for filters with raw filter rule schema
+ */
+export const filtersSchema = z.object({
+    type: z.enum(['and', 'or']).describe('Type of filter group operation'),
+    dimensions: z.array(filterRuleSchema).nullable(),
+    metrics: z.array(filterRuleSchema).nullable(),
+});
```

Comment:
this is the raw raw schema - see filterRuleSchema vs filterRuleSchemaTransformed